<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <group ns="/simpleRobot">
        <param name="robot_description" command="$(find xacro)/xacro '$(find at_robot_simulation)/urdf/simpleRobot.xacro'" />

        <arg name="x" default="0"/>
        <arg name="y" default="0"/>
        <arg name="z" default="0.5"/>

        <node name="mybot_spawn" pkg="gazebo_ros" type="spawn_model" output="screen"
            args="-urdf -param robot_description -model simpleRobot -x $(arg x) -y $(arg y) -z $(arg z)" />

        <rosparam command="load" file="$(find at_robot_simulation)/config/joints.yaml" />

        <node name="controller_spawner" pkg="controller_manager" type="spawner"
            respawn="false" output="screen" ns="/simpleRobot"
            args="--namespace=/simpleRobot
            joint_state_controller
            left_wheel_controller
            right_wheel_controller
            --timeout 60">
        </node>

        <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />
        <node name="rqt_publisher" pkg="rqt_publisher"  type="rqt_publisher" />
    </group>
</launch>


